backend/routers/Pins.py

```
@@ -0,0 +1,45 @@
+
+    return {
+        "image_url": f"/static/pins/{filename}"
+    }
+
+@router.get("/", response_model=List[schemas.PinResponse])
+def read_pins(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
+    pins = db.query(models.Pin).offset(skip).limit(limit).all()
+    return pins
+
+@router.post("/", response_model=schemas.PinResponse)
+def create_pin(pin: schemas.PinCreate, db: Session = Depends(get_db)):
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)